### PR TITLE
Ticket 11 - post comment on article

### PR DESCRIPTION
**Pull request by Brynster99**

### CODE REVIEW

File: controllers/controllers.js [hunk 1]
```diff
@@ -55,10 +58,17 @@ exports.getArticleComments = (req, res, next) => {
...
-    .then((article) => {
-      res.status(200).send({ article });
-    })
+    .then((article) => res.status(200).send({ article }))
+    .catch(next);
+};
+
+// DELETEs...
+exports.deleteCommentById = (req, res, next) => {
+  return checkExists('comments', 'comment_id', req.params.comment_id)
```
Comment (dav3rid) [inline controllers/controllers.js]: You may not need to explicitly check if the comment exists here - take a look at the rows property that comes back! If the array is empty, you already have your answer!

File: __tests__/endpoints.test.js [hunk 2]
```diff
@@ -11,6 +11,64 @@ afterAll(() => db.end());
...
+    return request(app)
+      .post('/api/articles/2/comments')
+      .send({ username: 'icellusedkars' })
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe('Bad Request');
+      });
+  });
+});
+
```
Comment (dav3rid) [inline __tests__/endpoints.test.js:81]: I think another 400 test would be good for when request body keys are missing entirely!

File: controllers/controllers.js [hunk 3]
```diff
@@ -12,6 +13,30 @@ const {
...
+  if (
+    !req.body.hasOwnProperty('username') ||
+    !req.body.hasOwnProperty('body')
+  ) {
+    next({ status: 400, msg: 'Bad Request' });
+  }
+
+  Promise.all([
+    checkExists('users', 'username', req.body.username),
+    checkExists('articles', 'article_id', req.params.article_id),
```
Comment (dav3rid) [inline controllers/controllers.js]: For this one - you again shouldn't need to check existence of user or article - the FOREIGN KEY constraint should kick in if you even try to insert

File: __tests__/endpoints.test.js (hunk 2 above)
Comment (Brynster99) [reply __tests__/endpoints.test.js:81]: done!

File: controllers/controllers.js (hunk 3 above)
Comment (Brynster99) [reply controllers/controllers.js]: Have switched to handling the PSQL error for this one

**Pull request merged**